haystack/preview/components/reader.py

```diff
@@ -0,0 +1,232 @@
+from pathlib import Path
+from typing import List, Optional, Tuple, Union
+import os
+from haystack.preview import component, Document, Answer, Pipeline
+from haystack.lazy_imports import LazyImport
+from haystack.nodes.prompt.invocation_layer.utils import get_task
+
+with LazyImport(message="Run 'pip install farm-haystack[inference]'") as torch_and_transformers_import:
+    from transformers import AutoModelForQuestionAnswering, AutoTokenizer
+    from tokenizers import Encoding
+    import torch
+
+
+@component
+class ExtractiveReader:
+    def __init__(
+        self,
+        model: Union[Path, str],
+        device: Optional[str] = None,
+        max_seq_length: int = 384,
+        top_k: int = 10,
+        stride: int = 128,
+    ) -> None:
+        torch_and_transformers_import.check()
+        self.model = str(model)
+        self.device = device
+        self.loaded = False
+        self.max_seq_length = max_seq_length
+        self.top_k = top_k
+        self.stride = stride
+        if os.path.exists(self.model):
+            return
+        try:
+            task = get_task(self.model)
+            if task != "question-answering":
+                raise ValueError(f"The provided model does not support question answering. Its intended use is {task}.")
+        except RuntimeError:
+            raise ValueError(
+                "The provided model either does not exist or it does not have a HF transformers compatible format."
+            )
```

Comment:
I am not really happy with this validation.
If it's a local model, we only check if the path exists. However, if it's not local, we check that the task in the config is correct.
Normally, you would think that doing this check for offline models should be easier. However, the HF library only supports this check for remote models which is why we have also solved it like this in the invocation layer (see #4760).